UCX 1.9.0: configure script insists on cray-ugni support

### Describe the bug
configure script insists on cray-ugni support even when --without-ugni is given on the configure command line.

### Steps to Reproduce
- ./configure --without-ugni
- UCX 1.9.0

### Setup and versions
Red Hat Enterprise Linux Server release 7.7 (Maipo)
Linux login4.ufhpc 3.10.0-1062.18.1.el7.x86_64 #1 SMP Wed Feb 12 14:08:31 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

./configure --build=x86_64-redhat-linux-gnu --host=x86_64-redhat-linux-gnu --program-prefix= --disable-dependency-tracking --prefix=/usr --exec-prefix=/usr --bindir=/usr/bin --sbindir=/usr/sbin --sysconfdir=/etc --datadir=/usr/share --includedir=/usr/include --libdir=/usr/lib64 --libexecdir=/usr/libexec --localstatedir=/var --sharedstatedir=/var/lib --mandir=/usr/share/man --infodir=/usr/share/info --disable-optimizations --disable-logging --disable-debug --disable-assertions --disable-params-check --with-ugni=no --enable-cma --without-cuda --without-gdrcopy --with-verbs --without-cm --without-knem --with-rdmacm --without-rocm --without-xpmem --enable-stats --enable-mt --with-avx --with-verbs --with-rc --with-ud --with-dc --with-rdmacm --with-mlx5-dv --with-ib-hw-tm --with-dm --without-cm --without-ugni
....
checking for CRAY_UGNI... no
configure: WARNING: UGNI support was requested but cray-ugni and cray-pmi packages can't be found
configure: error: Cannot continue

The patch does not resolve the issue.

commit b8b3caf11c9229f5ece82a8ba384a5f2e859a310
Author: Pavel Shamis (Pasha) <pasharesearch@gmail.com>
Date:   Sun Sep 27 12:16:45 2020 -0500

    Fix for https://github.com/openucx/ucx/issues/5731

-----------------------------------------------------------

+ ./configure --build=x86_64-redhat-linux-gnu --host=x86_64-redhat-linux-gnu --program-prefix= --disable-dependency-tracking --prefix=/usr --exec-prefix=/usr --bindir=/usr/bin --sbindir=/usr/sbin --sysconfdir=/etc --datadir=/usr/share --includedir=/usr/include --libdir=/usr/lib64 --libexecdir=/usr/libexec --localstatedir=/var --sharedstatedir=/var/lib --mandir=/usr/share/man --infodir=/usr/share/info --disable-optimizations --disable-logging --disable-debug --disable-assertions --disable-params-check --with-ugni=no --enable-cma --without-cuda --without-gdrcopy --with-verbs --without-cm --without-knem --with-rdmacm --without-rocm --without-xpmem --enable-stats --enable-mt --with-avx --with-verbs --with-rc --with-ud --with-dc --with-rdmacm --with-mlx5-dv --with-ib-hw-tm --with-dm --without-cm
checking for x86_64-redhat-linux-gnu-gcc... no
checking for gcc... gcc
checking whether the C compiler works... yes
...
checking for CRAY_UGNI... no
configure: WARNING: UGNI support was requested but cray-ugni and cray-pmi packages can't be found
configure: error: Cannot continue
error: Bad exit status from /var/tmp/rpm-tmp.IrSzsk (%build)


RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.IrSzsk (%build)

AFAICT, the patch does NOT fix the problem...

+ ./configure --build=x86_64-redhat-linux-gnu --host=x86_64-redhat-linux-gnu --program-prefix= --disable-dependency-tracking --prefix=/usr --exec-prefix=/usr --bindir=/usr/bin --sbindir=/usr/sbin --sysconfdir=/etc --datadir=/usr/share --includedir=/usr/include --libdir=/usr/lib64 --libexecdir=/usr/libexec --localstatedir=/var --sharedstatedir=/var/lib --mandir=/usr/share/man --infodir=/usr/share/info --disable-optimizations --disable-logging --disable-debug --disable-assertions --disable-params-check --with-ugni=no --enable-cma --without-cuda --without-gdrcopy --with-verbs --without-cm --without-knem --with-rdmacm --without-rocm --without-xpmem --enable-stats --enable-mt --with-avx --with-verbs --with-rc --with-ud --with-dc --with-rdmacm --with-mlx5-dv --with-ib-hw-tm --with-dm --without-cm
checking for x86_64-redhat-linux-gnu-gcc... no
checking for gcc... gcc
....
checking for rdma_create_id in -lrdmacm... yes
checking for CRAY_UGNI... no
configure: WARNING: UGNI support was requested but cray-ugni and cray-pmi packages can't be found
configure: error: Cannot continue

Did you re-run autogen.sh after applying the patch ?

On Tue, Sep 29, 2020 at 10:11 Charles A Taylor <notifications@github.com>
wrote:

>
>
> AFAICT, the patch does NOT fix the problem...
>
>
>
>
>
>    - ./configure --build=x86_64-redhat-linux-gnu
>    --host=x86_64-redhat-linux-gnu --program-prefix=
>    --disable-dependency-tracking --prefix=/usr --exec-prefix=/usr
>    --bindir=/usr/bin --sbindir=/usr/sbin --sysconfdir=/etc
>    --datadir=/usr/share --includedir=/usr/include --libdir=/usr/lib64
>    --libexecdir=/usr/libexec --localstatedir=/var --sharedstatedir=/var/lib
>    --mandir=/usr/share/man --infodir=/usr/share/info --disable-optimizations
>    --disable-logging --disable-debug --disable-assertions
>    --disable-params-check --with-ugni=no --enable-cma --without-cuda
>    --without-gdrcopy --with-verbs --without-cm --without-knem --with-rdmacm
>    --without-rocm --without-xpmem --enable-stats --enable-mt --with-avx
>    --with-verbs --with-rc --with-ud --with-dc --with-rdmacm --with-mlx5-dv
>    --with-ib-hw-tm --with-dm --without-cm
>
>
>    checking for x86_64-redhat-linux-gnu-gcc... no
>
>
>    checking for gcc... gcc
>
>
>    ....
>
>
>    checking for rdma_create_id in -lrdmacm... yes
>
>
>    checking for CRAY_UGNI... no
>
>
>    configure: WARNING: UGNI support was requested but cray-ugni and
>    cray-pmi packages can't be found
>
>
>    configure: error: Cannot continue
>
>
>
>
>
>
> —
> You are receiving this because you were assigned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/openucx/ucx/issues/5731#issuecomment-700772478>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AARB5WGBMXDF66ERUCRWELDSIH2JFANCNFSM4RXF3KNQ>
> .
>
>
>

--- a/src/uct/ugni/configure.m4
+++ b/src/uct/ugni/configure.m4
@@ -1,15 +1,14 @@
 #
 # Copyright (C) UT-Battelle, LLC. 2014. ALL RIGHTS RESERVED.
 # Copyright (C) Mellanox Technologies Ltd. 2001-2018.  ALL RIGHTS RESERVED.
+# Copyright (C) ARM Ltd. 2020.  ALL RIGHTS RESERVED.
 # See file LICENSE for terms.
 #
 
 cray_ugni_supported=no
 
 AC_ARG_WITH([ugni],
-            [AC_HELP_STRING([--with-ugni(=DIR)], [Build Cray UGNI support])],
-            [],
-            [with_ugni=guess])
+            [AC_HELP_STRING([--with-ugni(=DIR)], [Build Cray UGNI support])], [], [])
 
 AS_IF([test "x$with_ugni" != "xno"],
       [AC_MSG_CHECKING([cray-ugni])
@@ -20,12 +19,11 @@ AS_IF([test "x$with_ugni" != "xno"],
               uct_modules="${uct_modules}:ugni"
               cray_ugni_supported=yes
               AC_DEFINE([HAVE_TL_UGNI], [1], [Defined if UGNI transport exists])
-              ],
+             ],
              [AC_MSG_RESULT([no])
-              AS_IF([test "x$with_ugni" != "xguess"],
+              AS_IF([test "x$with_ugni" != "x"],
                     [AC_MSG_ERROR([UGNI support was requested but cray-ugni and cray-pmi packages cannot be found])])
-              ])
-       ])
+             ])])
 
 
 AM_CONDITIONAL([HAVE_CRAY_UGNI], [test "x$cray_ugni_supported" = xyes])

